### PR TITLE
ipn/localapi,client/tailscale,cmd/derper: add WhoIs lookup by nodekey, use in derper

### LINKED ISSUE #12465: cmd/tailscaled, derper: add cheaper derp --verify-clients lookup method
derper, when run with `--verify-clients`, currently does a full LocalAPI status fetch upon every DERP client's connection.

On small netmaps, this is fine.

On netmaps with 10k+ connections, it starts to look pretty inefficient. (the response payload JSON, despite coming from localhost with tons of bandwidth, is still a lot of JSON to parse)

We could add a cheaper LocalAPI call to look up by node key.

I'm thinking we overload WhoIs which already accepts two forms. We could add a third `nodekey:xxxxx` form.

### PR DESCRIPTION
**Pull request by bradfitz**
Fixes #12465

### CODE REVIEW

File: client/tailscale/localclient.go [hunk 1]
```diff
@@ -278,9 +283,31 @@ func decodeJSON[T any](b []byte) (ret T, err error) {
...
+	return decodeJSON[*apitype.WhoIsResponse](body)
+}
+
+// ErrPeerNotFound is returned by WhoIs and WhoIsNodeKey when a peer is not found.
+var ErrPeerNotFound = errors.New("peer not found")
+
+// WhoIsNodeKey returns the owner of the given wireguard public key.
+//
+// If not found, the error is ErrPeerNotFound.
+func (lc *LocalClient) WhoIsNodeKey(ctx context.Context, key key.NodePublic) (*apitype.WhoIsResponse, error) {
```
Comment (andrew-d) [inline client/tailscale/localclient.go:305]: Do we want to combine this with #12385 somehow?

Comment (bradfitz) [reply client/tailscale/localclient.go:305]: Seems unrelated?

Comment (bradfitz) [reply client/tailscale/localclient.go:305]: I mean, related only in that it's also called "whois". But the protocol doesn't apply to this lookup-by-nodekey.

Comment (andrew-d) [reply client/tailscale/localclient.go:305]: More in the "having a more generic signature on `LocalClient`" sense, but also happy to just keep separate 👍

**Review by raggi — APPROVED** [review]

**Pull request merged**